beps/0001-notifications-system/README.md

```diff
@@ -85,12 +85,43 @@ The role of the notifications plugin is to manage the lifecycle of notifications
 The notification backend stores notification using the [database service](https://backstage.io/docs/backend-system/core-services/index#database). In particular it needs to store the following information for each notification:
 
 - ID
-- Payload
+- Title
+- Priority
+- Origin
+- Topic (optional)
+- Creation timestamp (generated)
 - Receivers
 - Read status by user
+- Payload
+  - icon (optional)
+  - message (optional)
+  - links (list, optional)
 
 The receivers is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
 
+The title is mandatory human readable text shortly describing the notifications.
+
+The priority is one of the predefined values to help with presentation of the notifications or their filtering. Values: Critical, High, Normal, Low.
+
+The origin is a string identifying the creator of the notification, i.e. the BE plugin or external system.
+
+The topic is an optional string helping to group notifications of particular context. Its use cases include:
+
+- several notifications emitted by an asynchronous external task can be grouped by a single topic
+- a BE plugin can group several related messages to a particular processing, i.e. asynchronous progress monitoring
+
+The timestamp of message creation is auto-generated by the notifications backend at the time of receiving a request to create the notification.
+
+The icon is optional and is meant to improve UX. A string referencing an icon name from MUI icon library of a defined version. If missing, it will be determined from the priority
```

Comment:
We don't currently have all MUI icons in the frontend bundle so that's a technical challenge. We do have our icons API though, which likely ends up being paired up with some form of extension to build in select MUI icons using custom IDs in the new frontend system. i.e. perhaps not this exact implementation, but overall I think we'll make it work

Reply:
But for sure, this is just nice-to-have